test/QuadForm/Lattices.jl

```diff
@@ -11,17 +11,21 @@
   p = prime_decomposition(base_ring(L), 2)[1][1]
   @test @inferred islocally_isometric(L, M, p)
 
-  @inferred QuadLat(K, D)
   B = identity_matrix(K, 3)
   Bp = pseudo_matrix(B)
-  @inferred quadratic_lattice(K, Bp)
-  q = quadratic_lattice(K, Bp, gram=D);
-  @test_broken sprint(show, q)
+  # test lazy creation of the ambient space.
+  L = quadratic_lattice(K,Bp, gram=D)
+  @test dim(ambient_space(L)) == 3
+  @test sprint(show, L) isa String
+  @test gram_matrix(ambient_space(L)) == D
 
-  @inferred quadratic_lattice(K, B)
-  q = quadratic_lattice(K, B, gram=D);
-  @test_broken sprint(show, q)
+  L = quadratic_lattice(K,B, gram=D)
+  @test dim(ambient_space(L)) == 3
+  @test sprint(show, L) isa String
+  @test gram_matrix(ambient_space(L)) == D
 
+  B1 = pseudo_matrix(diagonal_matrix([a,a,a]))
+  @test_throws ArgumentError quadratic_lattice(K, B1, gram=D)
 
   fl = false
   while !fl
```

Comment:
This loop is now running forever apparently?

Reply:
let's see if L is integral the loop must terminate or there is a bug ... however if L is not integral, I can see it happen.